pkg/tlsutil/mtls_reload.go

```diff
@@ -1 +1,70 @@
 package tlsutil
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"sync"
+)
+
+type TLSReloader struct {
+	mu       sync.RWMutex
+	cert     *tls.Certificate
+	caPool   *x509.CertPool
+	certPath string
+	keyPath  string
+	caPath   string
+}
+
+func NewTLSReloader(certPath, keyPath, caPath string) (*TLSReloader, error) {
+	reloader := &TLSReloader{
+		certPath: certPath,
+		keyPath:  keyPath,
+		caPath:   caPath,
+	}
+	// reload works for a first time load as well
```

Comment:
just my thought i wanted to share, I reallllly dont need this to be changed
 one might want to create the TLSreloader before the files are on disk, then you cannot use this utility function, so my prefered style would not to have Reload withing NewTLSReloader.